doc/Language/traps.pod6

```diff
@@ -1990,5 +1990,22 @@ my @a = <1 2 3>;
 say @a.grep( 2 ~~ *.Int ); # OUTPUT: «(2)␤»
 =end code
 
+=head1 Closures
+
+Raku supports closures in many places. Not all of them are supposed to work.
+
+=head2 Role definition
+
+Roles can take a variable from the outer scope to initialize an attribute with a default value.
+
+=begin code
+my $seamingly-closure-var = 'will vanish';
+my $s = "don't do this" but role :: { has $.attr = $seamingly-closure-var }
+=end code
+
+The attribute will turn into C<Any> eventually. This trap is particular nasty because this may not
+happen before the program ends. Concurrency will trigger this behaviour often. At the time of writing
+Rakudo does not provide a warning either.
+
```

Comment:
But the example does not provide an actual working version of the trap. If it's going to turn into Any, it would be better if you provided an example where it does.

Reply:
We already got an example in /language/typesystem . Link will be provided.

Reply:
Not sure the link above is the one you intended to provide, but anyway it's much clearer now.